lib/engine/game/g_1848/step/tasmania_tile.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/special_track'
+
+module Engine
+  module Game
+    module G1848
+      module Step
+        class TasmaniaTile < G1848::Step::SpecialTrack
+          def actions(_entity)
+            return [] if !@game.private_closed_triggered || @game.tasmania.closed?
+
+            if @active_entity.nil?
+              @active_entity = @game.tasmania
+              @game.log << "#{@game.tasmania.name} must lay tasmania tile"
+            end
+
+            ['lay_tile']
+          end
+
+          def description
+            'Tasmania tile'
+          end
+
+          def blocks?
+            @game.tasmania
+          end
+
+          def active?
+            active_entities.any?
```

Comment:
!active_entities.empty?

Why? active_entities.any? loops through blocks and results in 100's of lines of JS. active_entities.empty? maps to array.length == 0 (1 line of code). Game load performance is significantly better when .empty? is used instead of .any?. The exception is when you need to pass in a block to any? as the block looping is then needed and .empty? won't work.